### PR TITLE
GATT Discovery Manager fix for nRF51

### PR DESCRIPTION
**Pull request by grochu**
Fixed data alignment on nRF51 (only aligned access is allowed
on Cortex M0).

@rakons, @lemrey could you review?

### CODE REVIEW

**Comment by grochu** [comment]
> Thank you, I have one question.
> Since the alignment is a compile time constant, would it not be better/possible
> to have `user_data_store` determine it by itself instead of receiving it as a parameter from the caller?

Well, yes, maybe it's cleaner if the `UUID_ALIGN` macro is used directly in the function instead of passing as parameter. I'll update the PR.

**Comment by grochu** [comment]
Hi @lemrey, @joerchan, I updated the PR as suggested.

File: subsys/bluetooth/gatt_dm.c [hunk 1]
```diff
@@ -15,6 +15,12 @@ LOG_MODULE_REGISTER(bt_gatt_dm, CONFIG_BT_GATT_DM_LOG_LEVEL);
 /* Available sizes: 128, 512, 2048... */
 #define CHUNK_SIZE (128 - sizeof(struct k_mem_block_id))
 
+#if __CORTEX_M == (0UL)
+	#define UUID_ALIGN 4U
```
Comment (lemrey) [inline subsys/bluetooth/gatt_dm.c]: nit: please don't indent preprocessor directives, unless they are complex macros.

**Comment by joerchan** [comment]
Thanks @grochu, I took the liberty of fixing lemreys nitpick comments.

**Pull request merged**